### ISSUE TITLE
Keep widgets width to height ratio on resize

### ISSUE THREAD

**Issue by Darkray-Fross** [troolee/gridstack.js#391]
I asked this question a week ago but it was closed before I got any satisfying answer so I will try again: As the title says I need a way to keep widgets width to height ratio always the same just like in angular-gridster plugin, but because gridstack has its width in percentage and height in pixels it changes its shape when you resize the window. Last time @troolee  suggested me to attach handler to window resize event and update cell height property of the grid but that don't seem to work, I tried it and nothing happens and by what value exactly do I update the height? Anyway, that doesn't seem to resolve my problem. I though a good way would be to add percentage height to gridstack functionality, but then it would be percentage of the parent which is gridstack instance which is in pixels, so that wouldn't work anyway.

**Comment by radiolips** [comment]
I apologize for your issue being closed before you were sufficiently satisfied with the answer.

Please provide an example where you have changed the height. You can listen to window resize and always resize both width and height. gridstack provides API calls for both. I'll be happy to help you get this code working.

If you need a more instant form of communication (though this is close), please feel free to join our Slack community, as mentioned on the landing page for this repo. We'll get this figured out!

**Comment by krema** [comment]
Are you using AngularJS? If yes you can watch the gridstack container width and adapt the defaultCellHeight. 

Example code:

    scope.$watch(
      function() {
        return angular.isDefined(gridstack) ? gridstack.container.width() : undefined;
      },
      function(newValue, oldValue) {
        if (gridstack && newValue && newValue != oldValue) {
          // 1000px is the reference width without need to adapt the zoom level
          var zoomLevel = defaultCellHeight * newValue / 1000; 
          gridstack.cellHeight(zoomLevel + defaultCellHeightUnit);
        }
      }
    );

**Comment by Darkray-Fross** [comment]
Thanks for the reply @radiolips , so I have somehow managed to make it work the way you and @troolee suggested, this is what my function look like ( I'm not sure if do it the proper way) : 

```
$( window ).on( 'resize', function() {
	var grid = $('.grid-stack').data( 'gridstack' );
	grid.cellHeight( grid.cellWidth() * 0.3 );
});
```

And that would resolve my problem if it wouldn't cause another. So let's say i have a widget and its responsive the way I want, then I want to make it wider so I'm resizing it's width ( with the ui-resizable-handle ), and the widget is doing what I want but the placeholder is also incrising its height ( obviously because I used cellWidth as a base of cellHeight ) then when I release the handle widget gets the size according to the placeholder, and then when I trigger the function I posted, the widget resizes to the actuall size I gave it ( when resizing with the ui-resizable-handle ) and finnaly works as it should. So maybe I should use something else instead of cellWidth?

@krema Nope, no angular, my knowledge of it is very limited, so I don't plan to use it, but if there is a way to do it in AngularJS then there is also a way to do it in VanillaJS ;)
